pennylane_lightning/core/src/simulators/lightning_qubit/gates/GateIndices.cpp

```diff
@@ -47,4 +47,34 @@ auto generateBitPatterns(const std::vector<std::size_t> &qubitIndices,
     }
     return indices;
 }
+
+/**
+ * @brief Introduce quantum controls in indices generated by
+ * generateBitPatterns.
+ *
+ * @param indices Indices for the operation.
+ * @param num_qubits Number of qubits in register.
+ * @param controlled_wires Control wires.
+ * @param controlled_values Control values (false or true).
+ */
+void controlBitPatterns(std::vector<std::size_t> &indices,
+                        const std::size_t num_qubits,
+                        const std::vector<std::size_t> &controlled_wires,
+                        const std::vector<bool> &controlled_values) {
+    constexpr std::size_t one{1U};
+    if (controlled_wires.empty()) {
+        return;
+    }
+    std::for_each(
```

Comment:
Nice! :star_struck:

Reply:
Could we update the indices in parallel if we used the standard for-loop instead?

 ``` c++
    PL_LOOP_PARALLEL(1)
    for (size_t i = 0; i < indices.size(); i++) {
    ...
```

Reply:
We could, but since `indices` has size `2**n_targets` (no controls) it should be smaller than 64 entries most of the time. Probably not worth spawning a pool of threads.